### PR TITLE
fix(NODE-5971): attach `v` to createIndexes command when `version` is specified

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

`createIndex` and `createIndexes` now support specifying an index version with the `version` option.

One note is that `createIndexes()` requires that users specify the `version` as a part of their index specification - a `version` provided as a part of the options is ignored.  This behavior is consistent with other index options provided to `createIndexes`.

#### What is the motivation for this change?

(release highlight left empty since `v` is an undocumented feature, no need for it to appear in our release notes)
### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/operations/indexes.ts [hunk 1]
```diff
@@ -212,16 +245,12 @@ export class CreateIndexesOperation extends CommandOperation<string[]> {
 
     this.options = options ?? {};
     this.collectionName = collectionName;
-    this.indexes = indexes.map(userIndex => {
+    this.indexes = indexes.map((userIndex: IndexDescription): ResolvedIndexDescription => {
       // Ensure the key is a Map to preserve index key ordering
       const key =
         userIndex.key instanceof Map ? userIndex.key : new Map(Object.entries(userIndex.key));
       const name = userIndex.name != null ? userIndex.name : Array.from(key).flat().join('_');
```
Comment (W-A-James) [inline src/operations/indexes.ts:252]: Optional, but while we're here
```suggestion
      const name = userIndex.name ?? Array.from(key).flat().join('_');
```

**Review by W-A-James — COMMENTED** [review]
Looks good, just one optional change

**Review by W-A-James — APPROVED** [review]

File: src/operations/indexes.ts [hunk 2]
```diff
@@ -193,14 +194,46 @@ function makeIndexSpec(
...
 }
 
+/**
+ * Receives an index description and returns a modified index description which has had invalid options removed
+ * from the description and has mapped the `version` option to the `v` option.
+ */
+function resolveIndexDescription(
+  description: IndexDescription
+): Omit<ResolvedIndexDescription, 'key'> {
+  const validProvidedOptions = Object.entries({ ...description }).filter(([optionName]) =>
```
Comment (nbbeeken) [inline src/operations/indexes.ts:207]:
```suggestion
  const validProvidedOptions = Object.entries(description).filter(([optionName]) =>
```

Isn't the spread redundant? Or did I miss something about the input type?

Comment (baileympearson) [reply src/operations/indexes.ts:207]: it's a speed-up loop: https://thedailywtf.com/articles/The-Speedup-Loop

Just joking 😄 this change should be safe, it was just copied from the original implementation.

**Review by W-A-James — APPROVED** [review]

**Pull request merged**